readme.md

```diff
@@ -1,28 +1,484 @@
+
+// vs
+
+let obj = {x,y}
+```
+
+#### You do: Concise methods and properties practice
+
+1. https://github.com/ga-wdi-exercises/es6-exercises/blob/master/07-concise-properties-and-methods.js
+
+### getters and setters
+
+Getters and setters in ES6 are very similar to getters and setters in Ruby classes,
+but allow us to define pseudo-properties on objects.
+
+Consider the following example:
+
+```js
+let person = {
+  firstName: "Jesse",
+  lastName: "Shawl",
+  get fullName(){
+    return this.firstName + " " + this.lastName
+  },
+  set fullName(fn){
+    var names = fn.split(" ")
+    this.firstName = names[0]
+    this.lastName = names[1]
+  }
+}
+person.fullName = "j dog" // notice no parentheses
+```
+
+#### You do: Getters and Setters
+
+1. https://github.com/ga-wdi-exercises/es6-exercises/blob/master/08-getters-setters.js
+
+### template literals
+
+Remember string interpolation from ruby? We've been able to semi-accomplish this
+with string concatenation in javascript:
+
+```js
+var name = "Inigo Montoya"
+var killee = "father"
+var prepareTo = "die"
+
+console.log("Hello. My name is "+ name + ". You killed my " + killee +". Prepare to " + prepareTo)
+```
+
+In ES6, we can interpolate variables using template literal syntax: `\``
+
+```js
+let name = "Inigo Montoya"
+let killee = "father"
+let prepareTo = "die"
+
+console.log(`Hello. My name is ${name}. You killed my ${killee}. Prepare to ${prepareTo}`)
+
+```
+
+#### You do: Template Exercises
+
+1. https://github.com/ga-wdi-exercises/es6-exercises/blob/master/09-templates.js
+1. https://github.com/ga-wdi-exercises/es6-exercises/blob/master/10-templates.js
+
+### arrow functions
+
+Arrow functions are a new shorthand syntax for defining anonymous functions:
+
+```js
+let foods = ["pizza","mac n cheese","lasagna"]
+foods.forEach( food => console.log(`I love ${food}`) )
+
+// vs the old
+
+foods.forEach(function(food){
+  console.log("I love " + food)
+})
+```
+
+If there is more than one argument to the anonymous function, wrap
+them in parens:
+
+```js
+let foods = ["pizza","mac n cheese","lasagna"]
+foods.forEach( (food,i) => console.log(`My #${i} favorite food is ${food}`) )
+```
+
+Arrow functions also have the benefit of not changing the value of `this`:
+
+```js
+function Person(){
+  this.age = 0
+  setInterval(function(){
+    this.age++ // doesnt work because this is GLOBAL
+  }, 1000)
+}
+
+var bob = new Person()
+
+// vs ES6
+
+function Person(){
+  this.age = 0
+  setInterval(() => {
+    this.age++
+  }, 1000)
+}
+
+var bob = new Person()
+```
+
+#### You do: Arrow functions
+
+1. https://github.com/ga-wdi-exercises/es6-exercises/blob/master/11-arrow-functions.js
 
 ## Classes
 
+Classes in ES6 are defined as syntactical sugar on top of the existing prototypal
+interface we've worked with:
+
+```js
+function Animal(name){
+  this.name = name 
+}
+Animal.prototype.speak = function(){
+  console.log("my name is " + this.name)
+}
+
+function Dog(name){
+  this.name = name
```

Comment:
Does es6 class syntax eliminate the need to define a `name` property on the `Dog` class, like we had to do with the es5 example?
```js
function Dog (name) {
  this.name = name
}
```